### DISCUSSION TITLE
What are some example projects that bfe has implemented?

### DISCUSSION THREAD

**Discussion by curiosport** [bfenetworks/bfe#924] [Q&A (问题和回答)] [answered]
I want to use bfe in my project and although I already read its documentation I would like to see a real example of how it has been implemented, is there any open source that uses bfe that I can read?

**Comment by mileszhang2016** [comment, marked as answer]
BFE is a full-fledged layer-7 load balancer. It is NOT a library to be integrated into other systems (though httpdns server in Baidu is based on BFE open source project). 

Some typical scenario to use BFE:
1. BFE works as a layer-7 load balancer.
2. BFE works as a K8s ingress
   please refer to https://github.com/bfenetworks/ingress-bfe

If you are a Chinese reader, this book is helpful.
https://github.com/baidu/bfe-book

**Reply by curiosport** [reply]
So even if someone releases an Open Source project the BFE configuration will often be omitted, right?

**Reply by mileszhang2016** [reply]
If a project integrated BFE, it should applied the BFE configuration as required.
BFE configurations include BASIC configuration and configuration for BFE modules.
If some BFE modules are not used in the project,  the configuration for these BFE modules can be omitted.
But the BASIC configuration and configuration for BFE modules applied are still required.

btw, the application of  BFE to other open source project is welcomed.

**Reply by curiosport** [reply]
Ok, thanks for the information.
